File: androidproduk/app/src/main/java/com/example/produkapp/RegisterActivity.java
package com.example.produkapp;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.android.volley.Request;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class RegisterActivity extends AppCompatActivity {
    EditText regUsername, regPassword;
    Button btnDaftar;

    String URL= ApiConfig.BASE_URL + "register.php";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_register);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        regUsername=findViewById(R.id.regUsername);
        regPassword=findViewById(R.id.regPassword);
        btnDaftar=findViewById(R.id.btnDaftar);

        btnDaftar.setOnClickListener(view -> {
            String username = regUsername.getText().toString().trim();
            String password = regPassword.getText().toString().trim();

            if (username.isEmpty()||password.isEmpty()){
                Toast.makeText(this, "Tidak boleh kosong",Toast.LENGTH_SHORT).show();
                return;
            }
            StringRequest request = new StringRequest(Request.Method.POST,URL,
                    response->{
                try {
                    JSONObject obj = new JSONObject(response);
                    Toast.makeText(this,obj.getString("message"),Toast.LENGTH_SHORT).show();
                    if (obj.getString("status").equals("success")){
                        finish();
                    }
                } catch (Exception e){
                    Toast.makeText(this,e.getMessage(),Toast.LENGTH_SHORT).show();
                }
                    },
                    error->Toast.makeText(this,error.toString(),Toast.LENGTH_SHORT).show()
                    ){
                protected Map<String, String> getParams(){
                    Map<String, String>params=new HashMap<>();
                    params.put("username", username);
                    params.put("password", password);
                    return params;
                }
            };
            Volley.newRequestQueue(this).add(request);
        });
    }
}

File: KopiNusantara/app/src/main/java/com/example/kopinusantara/RegisterActivity.kt
package com.example.kopinusantara

import android.content.Intent
import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.example.kopinusantara.api.ApiClient
import com.example.kopinusantara.model.LoginResponse
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class RegisterActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        val fullName = findViewById<EditText>(R.id.fullName)
        val email = findViewById<EditText>(R.id.email)
        val password = findViewById<EditText>(R.id.password)
        val btnRegister = findViewById<Button>(R.id.btnRegister)
        val tvLogin = findViewById<TextView>(R.id.tvLogin)

        btnRegister.setOnClickListener {

            val data = mapOf(
                "full_name" to fullName.text.toString(),
                "email" to email.text.toString(),
                "password" to password.text.toString()
            )

            ApiClient.instance.register(data)
                .enqueue(object : Callback<LoginResponse> {

                    override fun onResponse(
                        call: Call<LoginResponse>,
                        response: Response<LoginResponse>
                    ) {
                        if (response.isSuccessful) {
                            Toast.makeText(this@RegisterActivity, "Register berhasil", Toast.LENGTH_SHORT).show()

                            // balik ke login
                            startActivity(Intent(this@RegisterActivity, LoginActivity::class.java))
                            finish()
                        } else {
                            Toast.makeText(this@RegisterActivity, "Register gagal", Toast.LENGTH_SHORT).show()
                        }
                    }

                    override fun onFailure(call: Call<LoginResponse>, t: Throwable) {
                        Toast.makeText(this@RegisterActivity, t.message, Toast.LENGTH_SHORT).show()
                    }
                })
        }

        tvLogin.setOnClickListener {
            finish() // balik ke login
        }
    }
}